config/configtls/configtls.go

```diff
@@ -274,7 +277,14 @@ func (c TLSSetting) loadCert(caPath string) (*x509.CertPool, error) {
 		return nil, fmt.Errorf("failed to load CA %s: %w", caPath, err)
 	}
 
-	certPool := x509.NewCertPool()
+	var certPool *x509.CertPool
+	if c.IncludeSystemCACertsPool {
+		if certPool, err = x509.SystemCertPool(); err != nil {
+			return nil, err
+		}
```

Comment:
Maybe just being overly paranoid, but is there any code path where there isn't an error and the pool is still uninitialized?

Reply:
I'm so sorry to have to confirm that you do indeed sound strangely unhinged, but I can maybe check if certPool is nil and do something about it?

Reply:
Joke aside, thanks for the feedback, I have changed the code to be a bit more defensive.